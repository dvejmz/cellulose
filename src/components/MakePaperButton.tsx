import React from 'react';

interface MakePaperButtonProps {
  onClick: () => void;
}

const MakePaperButton: React.FC<MakePaperButtonProps> = (props: MakePaperButtonProps) => (
  <div className="make-paper" data-test-id="make-paper-button">
    <button className="make-paper__button" onClick={props.onClick}>Make Paper</button>
  </div>
);

export default MakePaperButton;
