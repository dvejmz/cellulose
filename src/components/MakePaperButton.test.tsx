import React from 'react';
import { shallow } from 'enzyme';
import MakePaperButton from './MakePaperButton';

describe('<MakePaperButton />', () => {
  let makePaperButton: any;

  beforeEach(() => {
    makePaperButton = shallow(<MakePaperButton onClick={() => {}} />);
  });

  it('renders ok', () => {
    expect(makePaperButton.text()).toBe('Make Paper');
  });
});
