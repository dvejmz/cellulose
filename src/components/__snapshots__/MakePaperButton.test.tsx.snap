// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MakePaperButton /> renders 1`] = `
<div
  className="make-paper"
  data-test-id="make-paper-button"
>
  <button
    className="btn btn-primary make-paper__button"
    onClick={[Function]}
  >
    Make Paper
  </button>
</div>
`;
